test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
```

Comment:
oh, for testing that retries work you create a proxy and upload through the proxy... Then have the proxy break the connection.

You can supply a custom `baseUrl` for the queue object and mock the queue API for testing this. But that might be a bit hard... If you removed the `queue.createArtifact` call from the retry cycle you could give a url (that goes through a local proxy) to the upload function in a unit test.

This kind of thing is hard to test in an integration test. Or you add a config option specifying a proxy to use.. so you can break the proxy... Easiest is to just do the unit test and observe that it retries when you give it an invalid URL :)